{
  try {
    super.close();
  }
 catch (  RuntimeException ex) {
    if (failOnClose) {
      throw ex;
    }
    logger.debug("MockDirectoryWrapper#close() threw exception",ex);
  }
}
